shell/src/main/java/alluxio/cli/fsadmin/pathconf/RemoveCommand.java

```diff
@@ -69,25 +78,43 @@ public void validateArgs(CommandLine cl) throws InvalidArgumentException {
   @Override
   public int run(CommandLine cl) throws IOException {
     AlluxioURI path = new AlluxioURI(cl.getArgs()[0]);
+    boolean recursive = cl.hasOption(RECURSIVE_OPTION.getOpt());
+    Set<PropertyKey> keysToRemove = new HashSet<>();
     if (cl.hasOption(KEYS_OPTION_NAME)) {
       String[] keys = cl.getOptionValue(KEYS_OPTION_NAME).split(",");
-      Set<PropertyKey> propertyKeys = new HashSet<>();
       for (String key : keys) {
-        propertyKeys.add(PropertyKey.fromString(key));
+        keysToRemove.add(PropertyKey.fromString(key));
       }
-      mMetaConfigClient.removePathConfiguration(path, propertyKeys);
-    } else {
-      mMetaConfigClient.removePathConfiguration(path);
+    }
+    Set<AlluxioURI> pathsToRemove = recursive ? mMetaConfigClient.getConfiguration(
+        GetConfigurationPOptions
+            .newBuilder()
+            .setIgnoreClusterConf(true)
+            .build())
+        .getPathConf().keySet().stream()
+        .map(k -> new AlluxioURI(k))
+        .filter(k -> isAncestorOf(path, k))
+        .collect(Collectors.toSet()) : Collections.singleton(path);
+
+    if (keysToRemove.isEmpty()) {
+      for (AlluxioURI p : pathsToRemove) {
+        mMetaConfigClient.removePathConfiguration(p);
+      }
+      return 0;
+    }
+    for (AlluxioURI p : pathsToRemove) {
+      mMetaConfigClient.removePathConfiguration(p, keysToRemove);
     }
     return 0;
   }
 
   @Override
   public String getUsage() {
-    return String.format("%s [--%s <key1,key2,key3>] <path>%n"
-        + "\t--%s: %s",
+    return String.format("%s [-R/--recursive] [--%s <key1,key2,key3>] <path>%n"
+        + "\t--%s: %s" + "\t--%s: %s",
         getCommandName(), KEYS_OPTION_NAME,
-        KEYS_OPTION_NAME, KEYS_OPTION.getDescription());
+        KEYS_OPTION_NAME, KEYS_OPTION.getDescription(),
+        RECURSIVE_OPTION, RECURSIVE_OPTION.getDescription());
```

Comment:
`RECURSIVE_OPTION` is an `Option` not a `String`.

```suggestion
        RECURSIVE_OPTION.getOpt(), RECURSIVE_OPTION.getDescription());
```